Presentations/06-R-programming-practices.Rmd

```diff
@@ -932,24 +883,23 @@ function_name <- function(input) {
     * The command to obtain the mean of a vector is `mean(x)`
     * The command to get the SD of a vector is `sd(x)`
     * R is vectorized: you can operate vectors and number directly and the result will be a vector
+    * Don't forget to include the argument `na.rm = TRUE` in `mean()` and `sd()`
 
 ---
 
 # Custom functions
 
 ```{r}
-  zscore <- function(x) {
-    
-    mean <- mean(x, na.rm = T)
-    sd   <- sd(x, na.rm = T)
+zscore <- function(x) {
+    mean <- mean(x, na.rm = TRUE)
+    sd   <- sd(x, na.rm = TRUE)
     z    <- (x - mean)/sd
-    
     return(z)
-    
   }
+```
 
-  summary(zscore(whr$happiness_score))
-  
+```{r echo=FALSE}
+knitr::include_app("https://rrmaximiliano.shinyapps.io/learnr-app/")
 ```
```

Comment:
We can also add here, combining functions you made with the apply family.

Reply:
great idea, I added this in exercise 7 (slide 62) on ebdf5aa